Nexus unable to start after serive stop

After a gracefull stop I was unable to restart the nexus, the log contains the following messages:

```
50:57,977+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseManagerImpl - Configuring OrientDB pool component with per-core limit of 16
2019-10-09 12:50:57,987+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.repository.search.IndexSyncService - Rebuilding search indexes because database has diverged
2019-10-09 12:50:58,053+0000 ERROR [FelixStartLevel]  *SYSTEM org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl - Unable to restore BlobStore BlobStoreConfiguration{name='nexus-judong-blobstore', type='Google Cloud Storage', attributes={google cloud storage={bucket=nexus-judong-blobstore, credential_file=/home/nexus/judo-ng-nexus-blobstore.json}}}
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, org.sonatype.nexus.common.stateguard.InvalidStateException: Invalid state: NEW; allowed: [STARTED]
  at org.sonatype.nexus.blobstore.gcloud.internal.GoogleCloudBlobStore.<init>(GoogleCloudBlobStore.java:142)
  while locating org.sonatype.nexus.blobstore.gcloud.internal.GoogleCloudBlobStore
  while locating java.lang.Object annotated with *

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1028)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:126)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl$$EnhancerByGuice$$d3f8e568.CGLIB$start$19(<generated>)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl$$EnhancerByGuice$$d3f8e568$$FastClassByGuice$$26783bab.invoke(<generated>)
	at com.google.inject.internal.cglib.proxy.$MethodProxy.invokeSuper(MethodProxy.java:228)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:76)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:77)
	at com.google.inject.internal.InterceptorStackCallback.intercept(InterceptorStackCallback.java:55)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl$$EnhancerByGuice$$d3f8e568.start(<generated>)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl$$EnhancerByGuice$$7277f3a9.CGLIB$start$21(<generated>)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl$$EnhancerByGuice$$7277f3a9$$FastClassByGuice$$248a9e75.invoke(<generated>)
	at com.google.inject.internal.cglib.proxy.$MethodProxy.invokeSuper(MethodProxy.java:228)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:76)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at com.google.inject.internal.InterceptorStackCallback$InterceptedMethodInvocation.proceed(InterceptorStackCallback.java:77)
	at com.google.inject.internal.InterceptorStackCallback.intercept(InterceptorStackCallback.java:55)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl$$EnhancerByGuice$$7277f3a9.start(<generated>)
	at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
	at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
	at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:311)
	at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:208)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.sonatype.nexus.common.stateguard.InvalidStateException: Invalid state: NEW; allowed: [STARTED]
	at org.sonatype.nexus.common.stateguard.StateGuard._ensure(StateGuard.java:115)
	at org.sonatype.nexus.common.stateguard.StateGuard.access$1(StateGuard.java:108)
	at org.sonatype.nexus.common.stateguard.StateGuard$GuardImpl.run(StateGuard.java:271)
	at org.sonatype.nexus.common.stateguard.GuardedInterceptor.invoke(GuardedInterceptor.java:53)
	at org.sonatype.nexus.blobstore.gcloud.internal.GoogleCloudBlobStore.<init>(GoogleCloudBlobStore.java:147)
	at com.google.inject.internal.ProxyFactory$ProxyConstructor.newInstance(ProxyFactory.java:265)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:111)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:268)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1019)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1015)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:126)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	... 7 common frames omitted
2019-10-09 12:50:58,059+0000 ERROR [FelixStartLevel]  *SYSTEM org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl - Failed transition: NEW -> STARTED
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.track(BlobStoreManagerImpl.java:320)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:133)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
	at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
	at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:311)
	at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:208)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
	at java.lang.Thread.run(Thread.java:748)
2019-10-09 12:50:58,065+0000 ERROR [FelixStartLevel]  *SYSTEM org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl - Failed transition: NEW -> STARTED
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.track(BlobStoreManagerImpl.java:320)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:133)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
	at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
	at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:311)
	at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:208)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
	at java.lang.Thread.run(Thread.java:748)
2019-10-09 12:50:58,068+0000 ERROR [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusContextListener - Failed to start nexus
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.track(BlobStoreManagerImpl.java:320)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:133)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
	at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
	at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:311)
	at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:208)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
	at java.lang.Thread.run(Thread.java:748)
2019-10-09 12:50:58,074+0000 ERROR [FelixStartLevel]  *SYSTEM Felix - Framework listener delivery error.
java.lang.NullPointerException: null
	at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.track(BlobStoreManagerImpl.java:320)
	at org.sonatype.nexus.repository.internal.blobstore.BlobStoreManagerImpl.doStart(BlobStoreManagerImpl.java:133)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.repository.manager.internal.RepositoryManagerImpl.doStart(RepositoryManagerImpl.java:228)
	at org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.start(StateGuardLifecycleSupport.java:67)
	at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
	at org.sonatype.nexus.common.stateguard.StateGuard$TransitionImpl.run(StateGuard.java:193)
	at org.sonatype.nexus.common.stateguard.TransitionsInterceptor.invoke(TransitionsInterceptor.java:56)
	at org.sonatype.nexus.extender.NexusLifecycleManager.startComponent(NexusLifecycleManager.java:199)
	at org.sonatype.nexus.extender.NexusLifecycleManager.to(NexusLifecycleManager.java:111)
	at org.sonatype.nexus.extender.NexusContextListener.moveToPhase(NexusContextListener.java:311)
	at org.sonatype.nexus.extender.NexusContextListener.frameworkEvent(NexusContextListener.java:208)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1431)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
	at java.lang.Thread.run(Thread.java:748)
2019-10-09 12:50:58,128+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusContextListener - Uptime: 37 seconds and 765 milliseconds (nexus-oss-edition/3.19.0.01)
2019-10-09 12:50:58,129+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Shutting down
2019-10-09 12:50:58,132+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop SECURITY
2019-10-09 12:50:58,134+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop EVENTS
2019-10-09 12:50:58,135+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop SCHEMAS
2019-10-09 12:50:58,140+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop UPGRADE
2019-10-09 12:50:58,141+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop RESTORE
2019-10-09 12:50:58,142+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop STORAGE
2019-10-09 12:50:58,145+0000 INFO  [FelixStartLevel]  *SYSTEM org.ehcache.core.EhcacheManager - Cache 'NexusAuthenticatingRealm.authenticationCache' removed from EhcacheManager.
2019-10-09 12:50:58,148+0000 INFO  [FelixStartLevel]  *SYSTEM org.ehcache.core.EhcacheManager - Cache 'NexusAuthorizingRealm.authorizationCache' removed from EhcacheManager.
2019-10-09 12:50:58,150+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.cache.internal.ehcache.EhCacheManagerProvider - Cache-manager closed
2019-10-09 12:50:58,151+0000 INFO  [FelixStartLevel]  *SYSTEM org.elasticsearch.node - [02D2B718-1ED0955C-224C1282-CA9841E8-9B10A6C3] stopping ...
2019-10-09 12:50:58,241+0000 INFO  [FelixStartLevel]  *SYSTEM org.elasticsearch.node - [02D2B718-1ED0955C-224C1282-CA9841E8-9B10A6C3] stopped
2019-10-09 12:50:58,241+0000 INFO  [FelixStartLevel]  *SYSTEM org.elasticsearch.node - [02D2B718-1ED0955C-224C1282-CA9841E8-9B10A6C3] closing ...
2019-10-09 12:50:58,256+0000 INFO  [FelixStartLevel]  *SYSTEM org.elasticsearch.node - [02D2B718-1ED0955C-224C1282-CA9841E8-9B10A6C3] closed
2019-10-09 12:50:58,261+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseManagerImpl - Stopping 3 pools
2019-10-09 12:50:58,263+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseManagerImpl - Stopping pool: security
2019-10-09 12:50:58,264+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseManagerImpl - Stopping pool: component
2019-10-09 12:50:58,264+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseManagerImpl - Stopping pool: config
2019-10-09 12:50:58,265+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseServerImpl$1 - OrientDB Server is shutting down...
2019-10-09 12:50:58,266+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseServerImpl$1 - Shutting down protocols
2019-10-09 12:50:58,266+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.server.plugin.OServerPluginManager - Shutting down plugins:
2019-10-09 12:50:58,267+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.server.plugin.OServerPluginManager - - jmx
2019-10-09 12:50:58,268+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseServerImpl$1 - OrientDB Server shutdown complete
2019-10-09 12:50:58,269+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - Orient Engine is shutting down...
2019-10-09 12:50:58,270+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - - shutdown storage: component...
2019-10-09 12:50:58,318+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - - shutdown storage: security...
2019-10-09 12:50:58,343+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - - shutdown storage: config...
2019-10-09 12:50:58,382+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - - shutdown storage: OSystem...
2019-10-09 12:50:59,284+0000 INFO  [FelixStartLevel]  *SYSTEM com.orientechnologies.orient.core.Orient - OrientDB Engine shutdown complete
2019-10-09 12:50:59,285+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.internal.orient.DatabaseServerImpl - Shutdown
2019-10-09 12:50:59,286+0000 INFO  [FelixStartLevel]  *SYSTEM org.sonatype.nexus.extender.NexusLifecycleManager - Stop KERNEL
```

This is the second time it happens, there was not any error or immediate kill on the process. The storage settings havent't changed.
The plugin is compiled from the commit 91994c7ad03dee3c32955a224c0ee2808c8ae9ed and the Nexus 3 version is 3.19.0-01

I've also tried the rebuild of orientDB which doesn't helped.

```
java -jar /opt/nexus/lib/support/nexus-orient-console.jar
 
CONNECT plocal:/opt/sonatype-work/nexus3/db/component admin admin
 
export database component-export
drop database
create database plocal:/opt/sonatype-work/nexus3/db/component admin admin
import database component-export.json.gz
 
CONNECT plocal:/opt/sonatype-work/nexus3/db/component admin admin
 
REBUILD INDEX *
REPAIR DATABASE --fix-graph
REPAIR DATABASE --fix-links
REPAIR DATABASE --fix-ridbags
REPAIR DATABASE --fix-bonsai
DISCONNECT
```

Is there any idea? Is there any way to reconcile the store? (there are lot of artifacts there)

--- a/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStore.java
+++ b/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStore.java
@@ -137,8 +137,9 @@ public class GoogleCloudBlobStore
   @Inject
   public GoogleCloudBlobStore(final GoogleCloudStorageFactory storageFactory,
                               final BlobIdLocationResolver blobIdLocationResolver,
-                              final GoogleCloudDatastoreFactory datastoreFactory,
                               final PeriodicJobService periodicJobService,
+                              final ShardedCounterMetricsStore metricsStore,
+                              final GoogleCloudDatastoreFactory datastoreFactory,
                               final DryRunPrefix dryRunPrefix,
                               final MetricRegistry metricRegistry,
                               final BlobStoreQuotaService quotaService,
@@ -146,11 +147,11 @@ public class GoogleCloudBlobStore
                               final int quotaCheckInterval)
   {
     super(blobIdLocationResolver, dryRunPrefix);
+    this.periodicJobService = periodicJobService;
     this.storageFactory = checkNotNull(storageFactory);
-    this.metricsStore = new ShardedCounterMetricsStore(blobIdLocationResolver, datastoreFactory, periodicJobService);
+    this.metricsStore = metricsStore;
     this.datastoreFactory = datastoreFactory;
     this.metricRegistry = metricRegistry;
-    this.periodicJobService = periodicJobService;
     this.quotaService = quotaService;
     this.quotaCheckInterval = quotaCheckInterval;
   }
@@ -178,7 +179,8 @@ public class GoogleCloudBlobStore
     wrapWithGauge("liveBlobsCache.evictionCount", () -> liveBlobs.stats().evictionCount());
     wrapWithGauge("liveBlobsCache.requestCount", () -> liveBlobs.stats().requestCount());
 
-    metricsStore.init(getBlobStoreConfiguration());
+    metricsStore.setBlobStore(this);
+    metricsStore.start();
     this.quotaCheckingJob = periodicJobService.schedule(createQuotaCheckJob(this, quotaService, log), quotaCheckInterval);
   }
 

--- a/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStore.java
+++ b/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStore.java
@@ -126,7 +126,13 @@ public class GoogleCloudBlobStore
 
   private MetricRegistry metricRegistry;
 
+  private PeriodicJobService periodicJobService;
+
+  private BlobStoreQuotaService quotaService;
+
   private PeriodicJob quotaCheckingJob;
+
+  private final int quotaCheckInterval;
   
   @Inject
   public GoogleCloudBlobStore(final GoogleCloudStorageFactory storageFactory,
@@ -144,8 +150,9 @@ public class GoogleCloudBlobStore
     this.metricsStore = new ShardedCounterMetricsStore(blobIdLocationResolver, datastoreFactory, periodicJobService);
     this.datastoreFactory = datastoreFactory;
     this.metricRegistry = metricRegistry;
-    this.quotaCheckingJob = periodicJobService.schedule(createQuotaCheckJob(this, quotaService, log), quotaCheckInterval);
-
+    this.periodicJobService = periodicJobService;
+    this.quotaService = quotaService;
+    this.quotaCheckInterval = quotaCheckInterval;
   }
 
   @Override
@@ -172,6 +179,7 @@ public class GoogleCloudBlobStore
     wrapWithGauge("liveBlobsCache.requestCount", () -> liveBlobs.stats().requestCount());
 
     metricsStore.init(getBlobStoreConfiguration());
+    this.quotaCheckingJob = periodicJobService.schedule(createQuotaCheckJob(this, quotaService, log), quotaCheckInterval);
   }
 
   @Override

--- a/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStore.java
+++ b/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStore.java
@@ -136,7 +136,10 @@ public class ShardedCounterMetricsStore
         .setNamespace(namespace)
         .setKind(METRICS_STORE)
         .newKey(1L);
+  }
 
+  @Override
+  public void doStart() {
     this.flushJob = periodicJobService.schedule(() -> flush(), FLUSH_FREQUENCY_IN_SECONDS);
   }
 

--- a/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStore.java
+++ b/src/main/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStore.java
@@ -24,10 +24,15 @@ import java.util.concurrent.ConcurrentLinkedDeque;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.StreamSupport;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+
 import org.sonatype.nexus.blobstore.BlobIdLocationResolver;
 import org.sonatype.nexus.blobstore.api.BlobId;
+import org.sonatype.nexus.blobstore.api.BlobStore;
 import org.sonatype.nexus.blobstore.api.BlobStoreConfiguration;
 import org.sonatype.nexus.blobstore.api.BlobStoreMetrics;
+import org.sonatype.nexus.common.stateguard.Guarded;
 import org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport;
 import org.sonatype.nexus.scheduling.PeriodicJobService;
 import org.sonatype.nexus.scheduling.PeriodicJobService.PeriodicJob;
@@ -51,9 +56,12 @@ import com.google.datastore.v1.TransactionOptions;
 import com.google.datastore.v1.TransactionOptions.ReadOnly;
 import org.apache.commons.lang.StringUtils;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
 import static java.lang.String.format;
 import static org.sonatype.nexus.blobstore.gcloud.internal.DatastoreKeyHierarchy.NAMESPACE_PREFIX;
 import static org.sonatype.nexus.blobstore.gcloud.internal.DatastoreKeyHierarchy.NXRM_ROOT;
+import static org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.State.STARTED;
 
 /**
  * This duck type of NXRM blobstore MetricsStores intends to achieve cost efficient eventual consistency
@@ -84,6 +92,7 @@ import static org.sonatype.nexus.blobstore.gcloud.internal.DatastoreKeyHierarchy
  * This falls within Google Cloud's best practices for use, and results in a cheap and efficient way to store
  * blobstore metrics with high accuracy.
  */
+@Named
 public class ShardedCounterMetricsStore
     extends StateGuardLifecycleSupport
 {
@@ -115,11 +124,14 @@ public class ShardedCounterMetricsStore
 
   private String namespace;
 
+  private BlobStore blobStore;
+
   /**
    * @param locationResolver
    * @param datastoreFactory
    * @param periodicJobService
    */
+  @Inject
   public ShardedCounterMetricsStore(final BlobIdLocationResolver locationResolver,
                                     final GoogleCloudDatastoreFactory datastoreFactory,
                                     final PeriodicJobService periodicJobService) {
@@ -128,7 +140,15 @@ public class ShardedCounterMetricsStore
     this.periodicJobService = periodicJobService;
   }
 
-  public void init(final BlobStoreConfiguration configuration) throws Exception {
+  public void setBlobStore(final BlobStore blobStore) {
+    checkState(this.blobStore == null, "Do not initialize twice");
+    checkNotNull(blobStore);
+    this.blobStore = blobStore;
+  }
+
+  @Override
+  public void doStart() throws Exception {
+    BlobStoreConfiguration configuration = this.blobStore.getBlobStoreConfiguration();
     this.datastore = datastoreFactory.create(configuration);
     this.namespace = NAMESPACE_PREFIX + configuration.getName();
     this.shardRoot = datastore.newKeyFactory()
@@ -136,10 +156,6 @@ public class ShardedCounterMetricsStore
         .setNamespace(namespace)
         .setKind(METRICS_STORE)
         .newKey(1L);
-  }
-
-  @Override
-  public void doStart() {
     this.flushJob = periodicJobService.schedule(() -> flush(), FLUSH_FREQUENCY_IN_SECONDS);
   }
 
@@ -161,11 +177,14 @@ public class ShardedCounterMetricsStore
         .forEach( shard -> datastore.delete(shard));
     log.warn("Blobstore metrics data removed");
   }
+
+  @Guarded(by = STARTED)
   public void recordDeletion(final BlobId blobId, final long size) {
     String shard = getShardLocation(blobId);
     pending.add(new Mutation(shard, -size, -1L));
   }
 
+  @Guarded(by = STARTED)
   public void recordAddition(final BlobId blobId, final long size) {
     String shard = getShardLocation(blobId);
     pending.add(new Mutation(shard, size, 1L));
@@ -174,6 +193,7 @@ public class ShardedCounterMetricsStore
   /**
    * @return a {@link BlobStoreMetrics} containing the sums of size and count across all shards
    */
+  @Guarded(by = STARTED)
   public BlobStoreMetrics getMetrics() {
     Long count = getCount(COUNT);
     Long size = getCount(SIZE);
@@ -208,7 +228,7 @@ public class ShardedCounterMetricsStore
     }
   }
 
-  Entity getShardCounter(final String location) {
+  private Entity getShardCounter(final String location) {
     KeyFactory keyFactory = datastore.newKeyFactory().addAncestors(
         NXRM_ROOT,
         PathElement.of(METRICS_STORE, 1L)
@@ -231,7 +251,7 @@ public class ShardedCounterMetricsStore
     return datastore.put(entity);
   }
 
-  String getShardLocation(final BlobId blobId) {
+  private String getShardLocation(final BlobId blobId) {
     String location = locationResolver.getLocation(blobId);
     if (!location.contains("/")) {
       throw new IllegalArgumentException(
@@ -240,7 +260,7 @@ public class ShardedCounterMetricsStore
     return StringUtils.split(location, "/")[0];
   }
 
-  QueryResults<Entity> getShards() {
+  private QueryResults<Entity> getShards() {
     Query<Entity> shardQuery = Query.newEntityQueryBuilder()
         .setFilter(PropertyFilter.hasAncestor(shardRoot))
         .setNamespace(namespace)

--- a/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStoreIT.groovy
+++ b/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStoreIT.groovy
@@ -90,6 +90,8 @@ class GoogleCloudBlobStoreIT
 
   BlobStoreUsageChecker usageChecker = Mock()
 
+  ShardedCounterMetricsStore metricsStore
+
   def setup() {
     quotaService = new BlobStoreQuotaServiceImpl([
         (SpaceUsedQuota.ID): new SpaceUsedQuota()
@@ -99,11 +101,14 @@ class GoogleCloudBlobStoreIT
 
     log.info("Integration test using bucket ${bucketName}")
 
-    blobStore = new GoogleCloudBlobStore(storageFactory, blobIdLocationResolver, datastoreFactory,
-        periodicJobService, new DryRunPrefix("TEST "), metricRegistry, quotaService, 60)
-    blobStore.init(config)
+    metricsStore = new ShardedCounterMetricsStore(blobIdLocationResolver, datastoreFactory, periodicJobService)
 
+    blobStore = new GoogleCloudBlobStore(storageFactory, blobIdLocationResolver, periodicJobService, metricsStore,
+        datastoreFactory, new DryRunPrefix("TEST "), metricRegistry, quotaService, 60)
+    blobStore.init(config)
     blobStore.start()
+
+    metricsStore.start()
 
     usageChecker.test(_, _, _) >> true
   }
@@ -111,6 +116,8 @@ class GoogleCloudBlobStoreIT
   def cleanup() {
     blobStore.stop()
     blobStore.remove()
+    metricsStore.stop()
+    metricsStore.removeData()
   }
 
   def cleanupSpec() {
@@ -334,10 +341,13 @@ class GoogleCloudBlobStoreIT
       // make a second
       def bucket2 = "multi-tenancy-test-${uid}"
       def config2 = makeConfig("multi-tenant-test-${uid}", bucket2)
-      def blobStore2 = new GoogleCloudBlobStore(storageFactory, blobIdLocationResolver, datastoreFactory,
-          periodicJobService, new DryRunPrefix("TEST "), metricRegistry, quotaService, 60)
+      ShardedCounterMetricsStore metricsStore2 = new ShardedCounterMetricsStore(blobIdLocationResolver,
+          datastoreFactory, periodicJobService)
+      def blobStore2 = new GoogleCloudBlobStore(storageFactory, blobIdLocationResolver, periodicJobService, metricsStore2,
+          datastoreFactory, new DryRunPrefix("TEST "), metricRegistry, quotaService, 60)
       blobStore2.init(config2)
       blobStore2.start()
+      metricsStore2.start()
 
       // write one file to blobstore1
       byte[] data = new byte[256]
@@ -384,6 +394,8 @@ class GoogleCloudBlobStoreIT
     cleanup:
       blobStore2.stop()
       blobStore2.remove()
+      metricsStore2.stop()
+      metricsStore2.removeData()
       cleanupBucket(config2, bucket2)
   }
 

--- a/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStoreTest.groovy
+++ b/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/GoogleCloudBlobStoreTest.groovy
@@ -62,6 +62,8 @@ class GoogleCloudBlobStoreTest
 
   BlobStoreQuotaService quotaService = Mock()
 
+  ShardedCounterMetricsStore metricsStore = Mock()
+
   KeyFactory keyFactory = new KeyFactory("testing")
 
   def blobHeaders = [
@@ -69,8 +71,8 @@ class GoogleCloudBlobStoreTest
       (BlobStore.CREATED_BY_HEADER): 'admin'
   ]
   GoogleCloudBlobStore blobStore = new GoogleCloudBlobStore(
-      storageFactory, blobIdLocationResolver, datastoreFactory, periodicJobService, new DryRunPrefix("TEST "),
-      metricRegistry, quotaService, 60)
+      storageFactory, blobIdLocationResolver, periodicJobService, metricsStore, datastoreFactory,
+      new DryRunPrefix("TEST "), metricRegistry, quotaService, 60)
 
   def config = new BlobStoreConfiguration()
 

--- a/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStoreIT.groovy
+++ b/src/test/java/org/sonatype/nexus/blobstore/gcloud/internal/ShardedCounterMetricsStoreIT.groovy
@@ -18,6 +18,7 @@ import java.util.stream.IntStream
 import org.sonatype.nexus.blobstore.BlobIdLocationResolver
 import org.sonatype.nexus.blobstore.DefaultBlobIdLocationResolver
 import org.sonatype.nexus.blobstore.api.BlobId
+import org.sonatype.nexus.blobstore.api.BlobStore
 import org.sonatype.nexus.blobstore.api.BlobStoreConfiguration
 import org.sonatype.nexus.scheduling.PeriodicJobService
 import org.sonatype.nexus.scheduling.PeriodicJobService.PeriodicJob
@@ -40,6 +41,8 @@ class ShardedCounterMetricsStoreIT
 
   GoogleCloudDatastoreFactory datastoreFactory = new GoogleCloudDatastoreFactory()
 
+  BlobStore blobStore = Mock()
+
   PeriodicJobService periodicJobService = Mock({
     schedule(_, _) >> new PeriodicJob() {
       @Override
@@ -59,11 +62,14 @@ class ShardedCounterMetricsStoreIT
     ]
   }
   def setup() {
+    blobStore.getBlobStoreConfiguration() >> config
     metricsStore = new ShardedCounterMetricsStore(blobIdLocationResolver, datastoreFactory, periodicJobService)
-    metricsStore.init(config)
+    metricsStore.setBlobStore(blobStore)
+    metricsStore.start()
   }
 
   def cleanup() {
+    metricsStore.stop()
     metricsStore.removeData()
   }
 

